src/types.ts

```diff
@@ -78,6 +78,7 @@ export interface SiweClientConfig {
   loginUrl: string
   clockUrl: string
   headers?: HeadersInit
+  timeout?: number
```

Comment:
I made this a constructor option instead of per method. I think we can use the same timeout across different endpoints. Any strong opinions to make this a per endpoint option?